### PR TITLE
Added openfst package

### PR DESCRIPTION
**Pull request by mcernak**
Added conda recipe for the latest OpenFst Version 1.6.1.

### CODE REVIEW

File: recipes/openfst/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  url: http://www.openfst.org/twiki/pub/FST/FstDownload/openfst-1.6.1.tar.gz
+  sha1: c623a39394158cc5abd2e54c6dd7934b07394440
+
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - gcc >=4.7  # [linux]
```
Comment (183amir) [inline recipes/openfst/meta.yaml]: remove this.

File: recipes/openfst/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    - python
+  run:
+    - gcc >=4.7  # [linux]
```
Comment (183amir) [inline recipes/openfst/meta.yaml]: remove this.

File: recipes/openfst/meta.yaml (hunk 1 above)
Comment (183amir) [inline recipes/openfst/meta.yaml]: Add `toolchain`

File: recipes/openfst/build.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#! /usr/bin/env bash
+
+if [ "`uname`" != "Darwin" ]; then
+  export CC=${PREFIX}/bin/gcc
+  export CXX=${PREFIX}/bin/g++
+fi
```
Comment (183amir) [inline recipes/openfst/build.sh]: remove this.

File: recipes/openfst/build.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
```
Comment (183amir) [inline recipes/openfst/build.sh]: `chmod +x ./configure`

File: recipes/openfst/build.sh [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+./configure --prefix=${PREFIX} || exit 1
```
Comment (183amir) [inline recipes/openfst/build.sh]: remove `|| exit 1`

File: recipes/openfst/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,33 @@
+package:
+  name: openfst
+  version: "0.1"
```
Comment (183amir) [inline recipes/openfst/meta.yaml]: Put openfst version.

**Comment by 183amir** [comment]
Now make sure the CI is green. Ask questions if you can't fix them.

**Comment by mcernak** [comment]
I added toolchain requirement, and the build now fails due to aclocal 
dependency

WARNING: 'aclocal-1.14' is missing on your system. You should only need 
it if you modified 'acinclude.m4' or 'configure.ac' or m4 files included 
by 'configure.ac'. The 'aclocal' program is part of the GNU Automake 
package:

So I added also the automake requirement, however, the error is not 
resolved. I try to fix it.

Milos

**Comment by 183amir** [comment]
Maybe you need `autoconf`

File: recipes/openfst/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - toolchain
+    - automake
+    - python
```
Comment (183amir) [inline recipes/openfst/meta.yaml]: Is `python` needed?

File: recipes/openfst/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - toolchain
+    - autoconf
+    - python
+  run:
+    - toolchain
```
Comment (183amir) [inline recipes/openfst/meta.yaml]: `toolchain` should not be a run requirement.

File: recipes/openfst/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+test:
+  requires:
+    - toolchain
```
Comment (183amir) [inline recipes/openfst/meta.yaml]: Can you remove it from test for now? If the tests fail we might want to put it back.

**Comment by 183amir** [comment]
You might want to take a look at here too: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=openfst

**Comment by 183amir** [comment]
Myabe you need to run `./configure.ac` first?

**Comment by mcernak** [comment]
The CircleCI test fails with build dependencies. Maybe the test uses 
automake 1.15 (the last version), while OpenFST requires 1.14. I tried 
to specify it in the meta file but it did not help..

Milos

**Comment by 183amir** [comment]
Like I said, can you run ./configure.a.c before ./configure?

**Comment by mcernak** [comment]
./configure.ac is not a script, it is used by autoconf to create 
./configure:

https://www.gnu.org/software/autoconf/manual/autoconf-2.68/html_node/autoconf-Invocation.html

This is what I call in build.sh

**Pull request closed without merging**